### PR TITLE
Replaced obsolete JEP-12 with JEP-12-01.

### PR DESCRIPTION
**Pull request by springcomp**
Addresses #133.

### CODE REVIEW

**Comment by springcomp** [comment]
@gibson042 for your consideration.

File: jep-012-raw-string-literals.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 | **Author** | Michael Downling
 | **Status** | accepted
 | **Created**| 09-Apr-2015
+| **Obsoleted By**| [JEP-12-01](./jep-012-01-raw-string-literals.md)
```
Comment (gibson042) [inline jep-012-raw-string-literals.md]: This PR should update the README to document JEP naming conventions. I don't think the JEP-originalNumber-updateNumber pattern is bad, but it is non-obvious.

File: jep-012-01-raw-string-literals.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+`bar` -> Parse error/warning (implementation specific)
+```
+
+## Motivation
+
+Raw string literals are provided in [various programming languages](https://en.wikipedia.org/wiki/String_literal#Raw_strings) in order to prevent
+language specific interpretation (i.e., JSON parsing) and remove the need for
+escaping, avoiding a common problem called [leaning toothpick syndrome (LTS)](https://en.wikipedia.org/wiki/Leaning_toothpick_syndrome). Leaning toothpick
+syndrome is an issue in which strings become unreadable due to excessive use of
+escape characters in order to avoid delimiter collision (e.g., `\\\\\\\\\\\\`).
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]: You probably want to pull forward changes from https://github.com/jmespath-community/jmespath.spec/pull/140/files#diff-61454b0c01e81f0be5cfa14e8a86d2b73b7d3e79cba8a747a9c4e7621368370c .

File: jep-012-01-raw-string-literals.md [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+number of backslash characters doubles.
+
+
+3. Introduces an ambiguous rule to the JMESPath grammar that requires a prose
+based specification to resolve the ambiguity in parser implementations.
+
+The relevant literal grammar rules are currently defined as follows:
+
+```abnf
+literal = "`" json-value "`"
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]: Hmm, this presents an interesting challenge because the actual current state of https://github.com/jmespath-community/jmespath.spec/blob/e1a690900a83f98f7caf42b71e87e1c082a42e74/GRAMMAR.ABNF has since diverged (including by adoption of JEP-12 itself). It might be better to instead introduce an entirely new JEP number (rather than JEP-12-01) that still obsoletes JEP-12 but applies against the current state rather than this historical one. It would therefore be quite small, just confirming the syntax and semantics of raw string literals (including our recent changes regarding control characters) but removing support for legacy unquoted backtick strings.

The idea should be that JEP-12 becomes interesting only for historical purposes and is not necessary for a correct implementation or comprehension of the spec, but the new JEP contains no history other than what is necessary to explain itself in the context of a specification that has already incorporated JEP-12.

Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: The _currently defined_ sentence implicitely refers to the *then current jmespath.org specification* at the time.
I think it’s better to improve the wording to that effect.

As explained in another issue, it appears that JEPs have not been authored chronologically and some have been authored retroactively, in an effort to document a trail of changes for historical reasons, much like this issue is trying to do for JEP-12.

File: jep-012-01-raw-string-literals.md (hunk 2 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: 👍

File: jep-012-01-raw-string-literals.md (hunk 3 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: https://github.com/jmespath-community/jmespath.spec/pull/138/commits/da3a486a82bdafe97e3bdca63420b5ff56cb2980

File: README.md [hunk 4]
```diff
@@ -36,6 +36,13 @@ Then open a [discussion](https://github.com/jmespath-community/jmespath.spec/dis
 alternatives. Once the discussion has reached a mature level of feedback it will be assigned a JEP#. At this point it is safe to begin composing a
 JEP, along with all the required changes and documentation.
 
+In rare circumstances, JEPs will be superceded by newer versions of a JEP. Those
+updated JEPs will have a sequence number appended as a suffix to the original JEP#.
+
+For instance, [JEP-12](./jep-012-raw-string-literals.md) introduced `raw-string`
+literals but the grammar defined there did not match that which was eventually specified. An update has been published as
+[JEP-12-01](./jep-012-01-raw-string-literals.md).
```
Comment (gibson042) [inline README.md]: It's a bit odd that the first number has no leading zeroes but the second number does. I think it would make more sense to name the followup "JEP-12-1" or "JEP-12a"—the latter patterned after IEEE amendments (like how 802.11-1997 was updated by 802.11a, 802.11b, …, 802.11z [802.11-2012], 802.11aa, …, 802.11af [802.11-2016], …, 802.11ba) and having the benefit of lexicographically sorting after its predecessor.

File: jep-012-01-raw-string-literals.md [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+## Abstract
+
+This JEP proposes the following modifications to JMESPath in order to improve
+the usability of the language and ease the implementation of parsers:
+
+
+* Addition of a **raw string literal** to JMESPath that will allow expressions
+to contain raw strings that are not mutated by JSON escape sequences (e.g.,
+“\\n”, “\\r”, “\\u005C”).
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]: Should these be formatted as code literals?
```suggestion
`\n`, `\r`, `\u005C`).
```

File: jep-012-01-raw-string-literals.md [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+in many areas, but most notably when invoking functions or building up
+multi-select lists and hashes.
+
+The following expression returns the number of characters found in the string
+`"foo"`. When parsing this expression, `` `"foo"` `` is parsed as a JSON value
+which produces the string literal value of `foo`:
+
+```
+`"foo"`
+```
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
`````suggestion
The following expression produces the three-character string “foo” using a `` `…` `` JSON text literal:

```
`"foo"`
```
`````

File: jep-012-01-raw-string-literals.md [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+
+The following expression is functionally equivalent. Notice that the quotes are
+elided from the JSON literal:
+
+```
+`foo`
+```
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
`````suggestion
### Obsolete alternative

_This section recapitulates content from the original version of JEP-12 that is no longer accurate as of this replacement._

The following expression is functionally equivalent. Notice that the quotes are
elided from the JSON literal:

```
`foo`
```
`````

File: jep-012-01-raw-string-literals.md [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+actually want the data to be represented as it was literally provided
+(which can lead to LTS). If the data being escaped was meant to be used
+along with another language that uses `\` as an escape character, then the
+number of backslash characters doubles.
+
+
+3. Introduces an ambiguous rule to the JMESPath grammar that requires a prose
+based specification to resolve the ambiguity in parser implementations.
+
+At the time of authoring the original version of this JEP, the relevant literal grammar rules were [then defined](https://github.com/jmespath-community/jmespath.py/commit/95a393e32e5aeb5f984cf2d7811fad75d4e0df95) as follows:
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]: With the above section header, this can be restored.
```suggestion
The relevant literal grammar rules are currently defined as follows:
```

File: jep-012-01-raw-string-literals.md [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+
+It is reasonable to assume that the most common use case for a JSON literal in
+a JMESPath expression is to provide a string value to a function argument or
+to provide a literal string value to a value in a multi-select list or
+multi-select hash. In order to make providing string values easier, it was
+decided that JMESPath should allow the quotes around the string to be elided.
+
+This proposal posits that allowing quotes to be elided when parsing JSON
+literals should be deprecated in favor of adding a proper string literal rule
+to JMESPath.
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
This proposal posits that allowing quotes to be elided when parsing JSON
literals should be prohibited in favor of the proper string literal syntax.
```

File: jep-012-01-raw-string-literals.md [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+
+## Specification
+
+A raw string literal is value that begins and ends with a single quote, does
+not interpret escape characters, and may contain escaped single quotes to
+avoid delimiter collision.
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
A raw string literal is value that begins and ends with a single quote
and preserves embedded backslashes except those used to escape
backslash or single quote characters.
```

File: jep-012-01-raw-string-literals.md [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+
+### Examples
+
+Here are several examples of valid raw string literals and how they are
+parsed:
+
+
+* A basic raw string literal, parsed as `foo bar`:
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
* A basic raw string literal representing the seven-character string “foo bar”:
```

File: jep-012-01-raw-string-literals.md [hunk 12]
```diff
@@ -0,0 +1,358 @@
...
+
+```
+'foo bar'
+```
+
+
+* An escaped single quote, parsed as `foo'bar`:
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
* A raw string literal with an escaped single quote, representing the seven-character string “foo'bar”:
```

File: jep-012-01-raw-string-literals.md [hunk 13]
```diff
@@ -0,0 +1,358 @@
...
+
+```
+'foo\'bar'
+```
+
+
+* An escaped backslash character, parsed as `foo\bar`:
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
* A raw string literal with an escaped backslash character, representing the seven-character string “foo\bar”:
```

File: jep-012-01-raw-string-literals.md [hunk 14]
```diff
@@ -0,0 +1,358 @@
...
+
+* A raw string literal that contains new lines:
+
+```
+'foo
+bar
+baz!'
+```
+
+The above expression would be parsed as a string that contains new lines:
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
The above expression represents the multi-line string:
```

File: jep-012-01-raw-string-literals.md [hunk 15]
```diff
@@ -0,0 +1,358 @@
...
+
+```
+foo
+baz
+bar!
+```
+
+
+* A raw string literal that contains escape characters,
+parsed as `foo\nbar`:
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
* A raw string literal with a non-special backslash character,
representing the eight-character string “foo\nbar”:
```

File: jep-012-01-raw-string-literals.md [hunk 16]
```diff
@@ -0,0 +1,358 @@
...
+                    preserved-escape /
+                    raw-string-escape
+preserved-escape = escape (
+                    %x00-26 /           ;  ␀ - & precedes U+0027 "'" apostrophe
+                    %x28-5B /           ; ( -[ precedes U+005C "\" reverse solidus
+                    %x5D-10FFFF)        ; ] - ... "]" and all following code points
+raw-string-escape = escape (
+                    "'" /               ; \ apostrophe U+0027
+                    escape)             ; \ reverse solidus U+005C
+
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
raw-string = "'" *raw-string-char "'"
raw-string-char = (%x00-26 /            ; ␀ through '&' (precedes U+0027 APOSTROPHE "'")
                    %x28-5B /           ; '(' through '[' (precedes U+005C REVERSE SOLIDUS '\')
                    %x5D-10FFFF) /      ; ']' and all following code points
                    preserved-escape /
                    raw-string-escape
preserved-escape = escape (
                    %x00-26 /           ;  ␀ through '&' (precedes U+0027 APOSTROPHE "'")
                    %x28-5B /           ; '(' through '[' (precedes U+005C REVERSE SOLIDUS '\')
                    %x5D-10FFFF)        ; ']' and all following code points
raw-string-escape = escape ("'" / escape)

```

File: jep-012-01-raw-string-literals.md [hunk 17]
```diff
@@ -0,0 +1,358 @@
...
+```
+
+This rule allows any character inside of a raw string, including an escaped single quote or backslash.
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
This rule allows any character inside of a raw string,
including control characters and escaped single quotes or backslashes.
```

File: jep-012-01-raw-string-literals.md [hunk 18]
```diff
@@ -0,0 +1,358 @@
...
+
+In addition to adding a `raw-string` rule, the `literal` rule in the ABNF
+will be simplified to become:
+
+```
+literal = "`" json-value "`"
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
literal = "`" json-text "`"
```

File: jep-012-01-raw-string-literals.md [hunk 19]
```diff
@@ -0,0 +1,358 @@
...
+```
+
+## Impact
+
+The impact to existing users of JMESPath is that the use of a JSON literal
+in which the quotes are elided SHOULD be converted to use the string-literal
+rule of the grammar. Whether or not this conversion is absolutely necessary
+will depend on the specific JMESPath implementation.
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
The impact to existing users of JMESPath is that the use of a JSON literal
in which the quotes are elided MUST be quoted or converted to use the
raw-string rule of the grammar.
```

File: jep-012-01-raw-string-literals.md [hunk 20]
```diff
@@ -0,0 +1,358 @@
...
+
+Implementations MAY choose to support the old syntax of allowing elided quotes
+in JSON literal expressions. If an implementation chooses this approach, the
+implementation SHOULD raise some kind of warning to the user to let them know
+of the deprecation and possible incompatibility with other JMESPath
+implementations.
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
```

File: jep-012-01-raw-string-literals.md [hunk 21]
```diff
@@ -0,0 +1,358 @@
...
+
+In order to support this type of variance in JMESPath implementations, all of
+the JSON literal compliance test cases that involve elided quotes MUST be
+removed, and test cases regarding failing on invalid unquoted JSON values MUST
+not be allowed in the compliance test unless placed in a JEP 12 specific
+test suite, allowing implementations that support elided quotes in JSON
+literals to filter out the JEP 12 specific test cases.
```
Comment (gibson042) [inline jep-012-01-raw-string-literals.md]:
```suggestion
To accommodate legacy JMESPath implementations, all of
the JSON literal compliance test cases that involve elided quotes MUST be
removed, and test cases regarding failing on invalid unquoted JSON values MUST
not be allowed in the compliance test unless placed in a JEP 12 specific
test suite, allowing such implementations to filter them out.
```

**Review by gibson042 — CHANGES_REQUESTED** [review]

File: jep-012-01-raw-string-literals.md (hunk 5 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: I expect those to be formatted as JSON strings.

File: jep-012-01-raw-string-literals.md (hunk 7 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: Please, can you rephrase the `of of` as I do not know what you had in mind.

File: README.md (hunk 4 above)
Comment (springcomp) [reply README.md]: 👍

File: jep-012-01-raw-string-literals.md (hunk 5 above)
Comment (gibson042) [reply jep-012-01-raw-string-literals.md]: "JSON string" seems like a heavily ambiguous concept (it could mean "_JSON text_", or "_JSON text expressing an instance of the string type, [which necessarily begins and ends with `"`]_", or "_an arbitrary sequence of code points represented by JSON text expressing an instance of the string type_"). But regardless, the distinction seems important here, and not currently clear. Perhaps something like this?
```md
* Addition to JMESPath of a **raw string literal** that treats `\` as a
literal character in most cases and in particular does not interpret JSON
escape sequences (and can therefore directly represent strings such as
“\\n”, “\\r”, and “\\u005C” without transformation).
```

File: jep-012-01-raw-string-literals.md (hunk 7 above)
Comment (gibson042) [reply jep-012-01-raw-string-literals.md]: Done. It was intended to be "as of".

File: jep-012-01-raw-string-literals.md (hunk 5 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: Indeed I later reverted to use curly quotes for an abstract string and only used double-quotes in this specific paragraph that mentions sample escape séquences.

File: jep-012-01-raw-string-literals.md (hunk 7 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: Agreed. ✅

File: jep-012-01-raw-string-literals.md (hunk 8 above)
Comment (springcomp) [reply jep-012-01-raw-string-literals.md]: I’m not sure I follow. The grammar as written in JEP12a is not aligned with the current grammar.

File: jep-012a-raw-string-literals.md [hunk 22]
```diff
@@ -0,0 +1,351 @@
...
+
+## Abstract
+
+This JEP proposes the following modifications to JMESPath in order to improve
+the usability of the language and ease the implementation of parsers:
+
+
+* Addition of a **raw string literal** to JMESPath that will allow expressions
+to contain raw strings that are not mutated by JSON escape sequences (e.g.,
+`"\\n"`, `"\\r"`, `"\\u005C"`).
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]: I don't think double-quoted strings make sense in the paragraph introducing single-quoted strings.
```suggestion
* Addition of a **raw string literal** to JMESPath that will allow direct
expression of string contents that would otherwise be modified by
interpretation as JSON (e.g., `'\n'`, `'\r'`, `'\u005C'`).
```

File: jep-012a-raw-string-literals.md [hunk 23]
```diff
@@ -0,0 +1,351 @@
...
+* Deprecation of  the current literal parsing behavior that allows for unquoted
+JSON strings to be parsed as JSON strings, removing an ambiguity in the
+JMESPath grammar and helping to ensure consistency among implementations.
+
+This proposal seeks to add the following syntax to JMESPath:
+
+```
+'foobar'
+'foo\'bar'
+`bar` -> Parse error/warning (implementation specific)
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
`bar` -> Parse error
```

File: jep-012a-raw-string-literals.md [hunk 24]
```diff
@@ -0,0 +1,351 @@
...
+```
+
+## Motivation
+
+Raw string literals are provided in [various programming languages](https://en.wikipedia.org/wiki/String_literal#Raw_strings) in order to prevent
+language specific interpretation (i.e., JSON parsing) and remove the need for
+escaping, avoiding a common problem called [leaning toothpick syndrome (LTS)](https://en.wikipedia.org/wiki/Leaning_toothpick_syndrome). Leaning toothpick
+syndrome is an issue in which strings become unreadable due to excessive use of
+escape characters in order to avoid delimiter collision (e.g., `\\\\\\`).
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
escape characters in order to avoid delimiter collision (e.g., `"\\\\\\"`).
```

File: jep-012a-raw-string-literals.md [hunk 25]
```diff
@@ -0,0 +1,351 @@
...
+
+When evaluating a JMESPath expression, it is often necessary to utilize string
+literals that are not extracted from the data being evaluated, but rather
+statically part of the compiled JMESPath expression. String literals are useful
+in many areas, but most notably when invoking functions or building up
+multi-select lists and hashes.
+
+The following expression produces the three-character string `"foo"` using a `` `…` `` JSON text literal:
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
The following expression produces the three-character string “foo” using a `` `…` `` JSON text literal:
```

File: jep-012a-raw-string-literals.md [hunk 26]
```diff
@@ -0,0 +1,351 @@
...
+
+
+* A basic raw string literal, representing the seven-character string “foo bar”:
+
+```
+'foo bar'
+```
+
+
+* A raw string literal with an escaped single quote, representing the seven-character string “foo'bar”`:
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
* A raw string literal with an escaped single quote, representing the seven-character string “foo'bar”:
```

File: jep-012a-raw-string-literals.md [hunk 27]
```diff
@@ -0,0 +1,351 @@
...
+baz!'
+```
+
+The above expression represents the multi-line string:
+
+```
+foo
+baz
+bar!
+```
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
`````suggestion
* A raw string literal that contains new lines:

```
'foo
bar
baz!'
```

The above expression represents the multi-line string:

```
foo
bar
baz!
```
`````

File: jep-012a-raw-string-literals.md [hunk 28]
```diff
@@ -0,0 +1,351 @@
...
+'foo\nbar'
+```
+
+### ABNF
+
+The following ABNF grammar rules will be added, and is allowed anywhere an
+expression is allowed:
+
+```abnf
+raw-string = "'" *raw-string-char "'"
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
`````suggestion
The following ABNF grammar rules will be added:

```abnf
expression =/ raw-string
raw-string = "'" *raw-string-char "'"
`````

File: jep-012a-raw-string-literals.md [hunk 29]
```diff
@@ -0,0 +1,351 @@
...
+                    %x00-26 /           ;  ␀ - & precedes U+0027 "'" apostrophe
+                    %x28-5B /           ; ( -[ precedes U+005C "\" reverse solidus
+                    %x5D-10FFFF)        ; ] - ... "]" and all following code points
+raw-string-escape = escape (
+                    "'" /               ; \ apostrophe U+0027
+                    escape)             ; \ reverse solidus U+005C
+
+```
+
+This rule allows any character inside of a raw string,
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
These rules allow any character inside of a raw string,
```

File: jep-012a-raw-string-literals.md [hunk 30]
```diff
@@ -0,0 +1,351 @@
...
+The above expression represents the multi-line string:
+
+```
+foo
+baz
+bar!
+```
+
+
+* A raw string literal that contains escape characters, representing the eight-character string “foo\nbar”:
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
* A raw string literal that contains a preserved backslash character, representing the eight-character string “foo\nbar”:
```

File: jep-012a-raw-string-literals.md [hunk 31]
```diff
@@ -0,0 +1,351 @@
...
+raw-string = "'" *raw-string-char "'"
+raw-string-char = (%x00-26 /            ; ␀ - & precedes U+0027 "'" apostrophe
+                    %x28-5B /           ; ( - [ precedes U+005C "\" reverse solidus
+                    %x5D-10FFFF) /      ; ] - ... "]" and all following code points
+                    preserved-escape /
+                    raw-string-escape
+preserved-escape = escape (
+                    %x00-26 /           ;  ␀ - & precedes U+0027 "'" apostrophe
+                    %x28-5B /           ; ( -[ precedes U+005C "\" reverse solidus
+                    %x5D-10FFFF)        ; ] - ... "]" and all following code points
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
raw-string-char = (%x00-26 /            ; ␀ through '&' (precedes U+0027 APOSTROPHE "'")
                    %x28-5B /           ; '(' through '[' (precedes U+005C REVERSE SOLIDUS '\')
                    %x5D-10FFFF) /      ; ']' and all following code points
                    preserved-escape /
                    raw-string-escape
preserved-escape = escape (
                    %x00-26 /           ;  ␀ through '&' (precedes U+0027 APOSTROPHE "'")
                    %x28-5B /           ; '(' through '[' (precedes U+005C REVERSE SOLIDUS '\')
                    %x5D-10FFFF)        ; ']' and all following code points
```

File: jep-012a-raw-string-literals.md [hunk 32]
```diff
@@ -0,0 +1,351 @@
...
+raw-string-escape = escape (
+                    "'" /               ; \ apostrophe U+0027
+                    escape)             ; \ reverse solidus U+005C
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
                    "'" /               ; U+0027 APOSTROPHE "'"
                    escape)             ; U+005C REVERSE SOLIDUS '\'
```

File: jep-012a-raw-string-literals.md [hunk 33]
```diff
@@ -0,0 +1,351 @@
...
+The impact to existing users of JMESPath is that the use of a JSON literal
+in which the quotes are elided MUST be quoted or converted to use the
+raw-string rule of the grammar.
+
+In order to support this type of variance in JMESPath implementations, all of
+the JSON literal compliance test cases that involve elided quotes MUST be
+removed, and test cases regarding failing on invalid unquoted JSON values MUST
+not be allowed in the compliance test unless placed in a JEP 12 specific
+test suite, allowing implementations that support elided quotes in JSON
+literals to filter out the JEP 12 specific test cases.
```
Comment (gibson042) [inline jep-012a-raw-string-literals.md]:
```suggestion
To accommodate legacy JMESPath implementations, all of
the JSON literal compliance test cases that involve elided quotes MUST be
removed, and test cases regarding failing on invalid unquoted JSON values
MUST NOT be allowed in the compliance test unless placed in a JEP 12 specific
test suite, allowing such implementations to filter them out.
```

**Review by gibson042 — CHANGES_REQUESTED** [review]

File: jep-012a-raw-string-literals.md (hunk 25 above)
Comment (springcomp) [reply jep-012a-raw-string-literals.md]: Good catch.

**Comment by springcomp** [comment]
@gibson042 

Thank you very much for putting up with me and doing such an extensive review.
This is much appreciated.

I think we’re there now 🤞.

**Review by gibson042 — APPROVED** [review]
> I think we’re there now :crossed_fingers:.

I think so too!
